perf/runner/src/index.ts

```diff
@@ -0,0 +1,238 @@
+import { execSync } from 'child_process';
+import { versions } from './versions';
+import yargs from 'yargs';
+import fs from 'fs';
+import { BenchmarkResults, Benchmark, Result, IperfResults, PingResults, ResultValue } from './benchmark-result-type';
+
+async function main(clientPublicIP: string, serverPublicIP: string) {
+    const pings = runPing(clientPublicIP, serverPublicIP);
+    const iperf = runIPerf(clientPublicIP, serverPublicIP);
+
+    copyAndBuildPerfImplementations(serverPublicIP);
+    copyAndBuildPerfImplementations(clientPublicIP);
+
+    const benchmarks = [
+             runBenchmarkAcrossVersions({
+                 name: "Single Connection throughput – Upload 100 MiB",
+                 clientPublicIP,
+                 serverPublicIP,
+                 uploadBytes: 100 << 20,
+                 downloadBytes: 0,
+                 unit: "bit/s",
+                 iterations: 5,
+             }),
+             runBenchmarkAcrossVersions({
+                 name: "Single Connection throughput – Download 100 MiB",
+                 clientPublicIP,
+                 serverPublicIP,
+                 uploadBytes: 0,
+                 downloadBytes: 100 << 20,
+                 unit: "bit/s",
+                 iterations: 5,
+             }),
+             runBenchmarkAcrossVersions({
+                 name: "Connection establishment + 1 byte round trip latencies",
+                 clientPublicIP,
+                 serverPublicIP,
+                 uploadBytes: 1,
+                 downloadBytes: 1,
+                 unit: "s",
+                 iterations: 100,
+             }),
+    ];
+
+    const benchmarkResults: BenchmarkResults = {
+        benchmarks,
+        pings,
+        iperf,
+    };
+
+    // Save results to benchmark-results.json
+    fs.writeFileSync('./benchmark-results.json', JSON.stringify(benchmarkResults, null, 2));
+
+    console.error("== done");
+}
+
+function runPing(clientPublicIP: string, serverPublicIP: string): PingResults {
+    console.error(`= run 100 pings from client to server`);
+
+    let cmd = `ssh -o StrictHostKeyChecking=no ec2-user@${clientPublicIP} 'ping -c 100 ${serverPublicIP}'`;
```

Comment:
Not a priority, but it would be nice if we didn't make ourselves vulnerable to MITM attacks here. Is there any (easy) way to get the SSH fingerprint? I imagine this is a very common problem that people have when they programmatically boot up servers.

Reply:
Added to "Outstanding work for future pull requests:". Agree that this is not a priority.